### DISCUSSION TITLE
posts/using-statement-trap/

### DISCUSSION THREAD

**Discussion by giscus** [BYJRK/byjrk.github.io#3] [Announcements]
# posts/using-statement-trap/

using 语句的减少一层缩进的新用法在释放资源时可能会有意想不到的效果，不能无脑使用。

https://blog.coldwind.top/posts/using-statement-trap/

**Comment by Antecer** [comment]
竟然还有这种坑，又学到了新知识；然而下次写的时候仍然可能会被坑进去，毕竟VS会拼命建议你把花括号省略掉。

**Comment by washingtown** [comment]
感觉这个坑更多的来自GZipStream，输出结果取决于是否Dispose这一点正常不太会想到

**Reply by BYJRK** [reply]
是的，本质原因是它，毕竟其他绝大多数时候都可以无脑去掉花括号。归根结底还是要知道语法究竟对应的底层代码是怎样的
